### ISSUE TITLE
Build timings: I have to refresh the page to see startup script timing

### ISSUE THREAD

**Issue by bpmct** [coder/coder#15273]
https://github.com/user-attachments/assets/cdae3b10-bb87-43be-97e1-2dd48dea535f

**Comment by dannykopping** [comment]
cc @BrunoQuaresma

**Comment by BrunoQuaresma** [comment]
Hmm... it seems the timings are being added as the agent's lifecycle is set to ready. 🤔 I will try to figure out a better way to refresh the timings when the agent is ready.

**Comment by BrunoQuaresma** [comment]
To solve this issue, we need to define when the agent timings are ready to be displayed. I’m considering the following logic:

- The agent is marked as ready.
- If the agent has scripts, check that timings for each script are present.
- If the script timings aren’t present, continue refetching until they are available.

**Comment by dannykopping** [comment]
Could we perhaps always show the timings, and when there's an update: prompt the user to refresh?

Since this is such a long-running process, I think we need to be able to show intermediate phases. If we did the above we could simply poll, and if the state (i.e. timings JSON) has changed since the currently visible state then prompt. WDYT?

**Comment by BrunoQuaresma** [comment]
@dannykopping, it makes sense. @chrifro, do you have any ideas on how we could show or have a "pending" state for the chart? Or at least for the "run startup scripts"?

**Comment by chrifro** [comment]
>Could we perhaps always show the timings

With "timings" do you mean the data under "provisioning"?


Would this work?
![Image](https://github.com/user-attachments/assets/ef3dd05c-f861-4751-bf35-10fea59486b4)


What happens when new data is available? Can we load it automatically or do we need to reload the page?
![Image](https://github.com/user-attachments/assets/ea33cb98-4e40-402e-82dc-7358b9a6e28a)

See [Figma](https://www.figma.com/design/Zm4W96k6oawgS8gXIKKpk8/Provisioning-times?node-id=395-134&t=R7hmnjvJMLkmqcmg-1)

**Comment by BrunoQuaresma** [comment]
@chrifro and @dannykopping, what do you think about this? I believe this provides more context to the user regarding why it is still loading.

![Image](https://github.com/user-attachments/assets/4a63a6ba-fcb0-4e6c-9f88-89cabb11ad67)

**Comment by dannykopping** [comment]
@BrunoQuaresma looks good! The copy's grammar should be corrected to "Waiting for agent to become ready...".

**Comment by BrunoQuaresma** [comment]
When refetching the timings to progressively display the agent scripts, it causes a layout shift because the chart recalculates the best intervals to fit the bar sizes and inner elements. This issue looks strange and buggy. To avoid the layout shift, I see two options:

1. **Display the timings only when all timings are available:** This eliminates the need to recalculate intervals later.
2. **Calculate intervals and sizes only on the first render:** The downside is that without all the data, the chart may not determine the optimal sizes and intervals.

In my opinion, the first option is the most reasonable and safest. I suggest we move forward with it and, depending on user feedback, we can consider implementing the second option later. WDYT? @dannykopping @chrifro

**Comment by dannykopping** [comment]
Yeah I think the first option makes the most sense. You only really need to know the timing information retrospectively; there isn't much value in seeing it while the build is taking place.

**Comment by chrifro** [comment]
I'm also for option 1 🤝 ✨
